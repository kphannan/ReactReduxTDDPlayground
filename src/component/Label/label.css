.text-input-container {
     font-size: 16px;
     color:#999;
     margin-top: 0px;
     margin-bottom: 0px;

 }

.text-input-container-error {
    font-size: 16px;
    color: #ed1c24 !important;
    margin-top: 0px;
    margin-bottom: 0px;

}