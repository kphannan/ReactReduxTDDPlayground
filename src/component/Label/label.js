import React, { Component } from 'react'
import './label.css'

class Label extends Component {

    render () {
        const {className,id} = this.props
        return (
            <div className={className} id={id}>
                 <label>{this.props.children}</label>

            </div>
        )
    }
}

export default Label;