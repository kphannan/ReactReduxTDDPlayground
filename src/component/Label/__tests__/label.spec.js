/**
 * @jest-environment jsdom
 */

import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'
import Label from '../label'


Enzyme.configure({ adapter: new Adapter() })

describe('Label rendering Testing', () => {

    test('verify against label snapshot', () => {
        const component = shallow(<Label/>)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

    test("Should render Class Name ", ()=> {
        const component = shallow (<Label className="text-input-container-error" />)
        expect(component.prop('className')).toEqual('text-input-container-error');
    });

    test("Should render Owner Name inside a Label tag", ()=> {
        const component = shallow(<Label labelClassVal={"text-input-container-error"}>Owner Ldap</Label>)
        expect(component.prop('children').props.children).toEqual("Owner Ldap");
       expect(component.prop('children').type).toBe('label')

    })

})