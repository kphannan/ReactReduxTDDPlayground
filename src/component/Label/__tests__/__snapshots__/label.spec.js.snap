// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label rendering Testing verify against label snapshot 1`] = `
<div>
  <label />
</div>
`;
